src/materialized/src/bin/materialized/main.rs

```diff
@@ -807,6 +817,11 @@ For more details, see https://materialize.com/docs/cli#experimental-mode
     }
 }
 
+fn contains_duplicates(slice: &[String]) -> bool {
+    let dedup: HashSet<_> = slice.iter().collect();
+    dedup.len() != slice.len()
+}
```

Comment:
@petrosagg added a method to `Itertools` for this! https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.duplicates

```slice.iter().duplicates().next().is_some()```

Reply:
Wow, that's awesome :D I should learn this crate some time.

Reply:
Turns out there is even an [https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.all_unique](`all_unique`)!

Reply:
Even better! I looked for that, but under the name `has_duplicates` >.<.